esmvalcore/preprocessor/_cycles.py

```diff
@@ -0,0 +1,68 @@
+"""Operations related to cycles (annual cycle, diurnal cycle, etc.)."""
+import logging
+
+import iris
+import iris.coord_categorisation
+
+logger = logging.getLogger(__name__)
+
+
+def amplitude(cube, coords):
+    """Calculate amplitude of cycles by aggregating over coordinates.
+
+    Note
+    ----
+    The amplitude is calculated as `peak-to-peak` amplitude (difference
+    between maximum and minimum value of the signal). Other amplitude types
+    are currently not supported.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube
+        Input data.
+    coords : str or list of str
+        Coordinates over which is aggregated. For example, use ``'year'`` to
+        extract the annual cycle amplitude for each year in the data or
+        ``['day_of_year', 'year']`` to extract the diurnal cycle amplitude for
+        each individual day in the data. If the coordinates are not found in
+        ``cube``, try to add it via :mod:`iris.coord_categorisation` (this only
+        works for temporal coordinates).
+
+    Returns
+    -------
+    iris.cube.Cube
+        Amplitudes.
+
+    Raises
+    ------
+    iris.exceptions.CoordinateNotFoundError
+        A coordinate is not found in ``cube`` and cannot be added via
+        :mod:`iris.coord_categorisation`.
+
+    """
+    if isinstance(coords, str):
+        coords = [coords]
+
+    # Add coordinate if necessary
+    for coord_name in coords:
+        if cube.coords(coord_name):
+            continue
+        logger.debug("Trying to add coordinate '%s' to cube via iris."
+                     "coord_categorisation", coord_name)
+        if hasattr(iris.coord_categorisation, f'add_{coord_name}'):
+            getattr(iris.coord_categorisation, f'add_{coord_name}')(cube,
+                                                                    'time')
+            logger.debug("Added temporal coordinate '%s'", coord_name)
+        else:
+            raise iris.exceptions.CoordinateNotFoundError(
+                f"Coordinate '{coord_name}' is not a coordinate of cube "
+                f"{cube.summary(shorten=True)} and cannot be added via "
+                f"iris.coord_categorisation")
+
+    # Calculate amplitude
+    max_cube = cube.aggregated_by(coords, iris.analysis.MAX)
+    min_cube = cube.aggregated_by(coords, iris.analysis.MIN)
+    amplitude_cube = max_cube - min_cube
```

Comment:
nice, do you not want to care about the sign of the amplitude? ie this will always return a positive amplitude A that can come from two different (min, max) combinations depending on their signs

Reply:
What do you mean by *sign* of the amplitude? The order (min, max) / (max, min), so that `sin(x)` gives `+1` and `-sin(x)` gives `-1`?

Reply:
so in case of temperatures: diurnal amplitudes: say A = 10 since in January it goes from -1 to -11 and also A = 10 in July since it goes from 30 to 20; A = 10 in both cases - does that need a sign attached to it since A_jan is from two negative temps and A_jul is from two positive temps, or you don't care - you need an amplitude in the absolute meaning

Reply:
Ah, I see. For me an amplitude is always some kind of absolute distance, so there's no need for a sign.

Reply:
cool, we're set then! :beer: